lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -410,23 +395,34 @@ export class UserAgentApplication {
             return;
         }
 
-        // if extraScopesToConsent is passed in loginCall, append them to the login request
-        const scopes: Array<string> = isLoginCall ? this.appendScopes(request) : request.scopes;
-
-        // Validate and filter scopes (the validate function will throw if validation fails)
-        this.validateInputScope(scopes, !isLoginCall);
+        if (!request) {
+            request = {};
+        }
 
         // Get the account object if a session exists
-        const account: Account = (request && request.account && !isLoginCall) ? request.account : this.getAccount();
+        // const account: Account = (request && request.account && !isLoginCall) ? request.account : this.getAccount();
+
+        // Retrieve authority from request object, or use configured authority
+        const acquireTokenAuthority = (!isLoginCall && request && request.authority) ? AuthorityFactory.CreateInstance(request.authority, this.config.auth.validateAuthority) : this.authorityInstance;
```

Comment:
This is modified to remove !isLoginCall() check in a new PR. Merge conflict may not be straightforward. Please check.